crates/iota-indexer/src/handlers/mod.rs

```diff
@@ -101,9 +101,11 @@ impl<T> CommonHandler<T> {
                         if cancel.is_cancelled() {
                             return Ok(());
                         }
-                        for (cp_seq, data) in tuple_chunk {
-                            unprocessed.insert(cp_seq, (cp_seq, data));
-                        }
+                        unprocessed.extend(
+                            tuple_chunk
+                                .into_iter()
+                                .map(|(cp_seq, data)| (cp_seq, (cp_seq, data))),
+                        );
```

Comment:
I think this is rather unnecessary. It is more verbose, and ends up in the same internal logic: https://doc.rust-lang.org/src/alloc/collections/btree/map.rs.html#2312

Reply:
fixed: https://github.com/iotaledger/iota/pull/6155/commits/1b4b1523ac55ac2684c05f1b6e39129866414989